src/app/snippet/components/snippet-filter/snippet-filter.component.ts

```diff
@@ -0,0 +1,91 @@
+import {Component, Input, OnInit} from '@angular/core';
+import {Router} from '@angular/router';
+
+import {Snippet} from '../../models/snippet';
+import {Label} from '../../models/label';
+
+@Component({
+  selector: 'app-snippet-filter',
+  templateUrl: 'snippet-filter.component.html',
+  styleUrls: ['snippet-filter.component.scss']
+})
+export class SnippetFilterComponent implements OnInit {
+
+  buttonBackground = 'linear-gradient(0.44turn, mediumpurple, purple, black)';
+  buttonText = 'Nouveau Snippet';
+
+  @Input()
+  snippets: Snippet[] = [];
+  @Input()
+  labels: Label[] = [];
+  total = 0;
+  favorites = 0;
+  unlabeled = 0;
+  labelsUsedBySnippets = new Map<string, number | undefined>();
```

Comment:
```suggestion
  labelCountMap = new Map<string, number | undefined>();
```
je trouve que c'est plus compréhensible tu crois pas ?